### PR TITLE
Sort routes by path, then verb. Closes #21.

### LINKED ISSUE #21: Fix route sorting
Routes should be sorted by path, then by HTTP verb

### PR DESCRIPTION
**Pull request by arb**

### CODE REVIEW

File: lib/setup.js [hunk 1]
```diff
@@ -36,6 +36,11 @@ var constructResponse = function (server) {
           path: route.path,
           plugin: route.public.realm.plugin || null
         };
+      }).sort(function (a, b) {
```
Comment (cjihrig) [inline lib/setup.js:48]: I don't think this is right. For example, the routes `GET /` and `GET /get_address` would be sorted as `GET /`, `GET /get_address`. But `POST /` and `GET /get_address` would be the opposite. I think the correct way would be to `localeCompare()` the path. If the result is 0, then `localeCompare()` the method. Also, `sort()` should return a negative number, 0, or positive number, not a boolean.

Comment (arb) [inline lib/setup.js:48]: The problem is the result of `localeCompare` will never be 0 on the paths; you can't have overlapping paths with hapi. That's why I added the method in with the path.

Agreed about the return type though and the `localeCompare`.

File: lib/setup.js [hunk 2]
```diff
@@ -36,6 +45,12 @@ var constructResponse = function (server) {
...
+        var pathCompare = a.path.localeCompare(b.path);
+        if (pathCompare === 0) {
+          return methodSort[a.method] - methodSort[b.method];
```
Comment (cjihrig) [inline lib/setup.js:51]: Should probably uppercase the methods, just in case. Also, what happens if a method sneaks in besides the ones we check for?

Comment (arb) [inline lib/setup.js:51]: You're already uppercasing the methods on line 44. Also, hapi throws if you try to use a method that doesn't exist.

Comment (cjihrig) [inline lib/setup.js:51]: Perfect

**Pull request merged**